4.8/windows/nanoserver/Dockerfile

```diff
@@ -31,16 +31,40 @@ RUN Invoke-WebRequest $('https://nodejs.org/dist/v{0}/node-v{0}-win-x64.zip' -f
     Expand-Archive node.zip -DestinationPath C:\ ; \
     Rename-Item -Path $('C:\node-v{0}-win-x64' -f $env:NODE_VERSION) -NewName 'C:\nodejs'
 
+ENV YARN_VERSION 0.27.5
+
+RUN @( \
+    '6A010C5166006599AA17F08146C2130DFD2497F5' \
+    ) | foreach { \
+      gpg --keyserver ha.pool.sks-keyservers.net --recv-keys $_ ; \
+    }
+
+RUN $asc =  $('yarn-v{0}.tar.gz.asc' -f $env:YARN_VERSION) ; \
+    $tarGz =  $('yarn-v{0}.tar.gz' -f $env:YARN_VERSION) ; \
+    Invoke-WebRequest $('https://yarnpkg.com/downloads/{0}/yarn-v{0}.tar.gz.asc' -f $env:YARN_VERSION) -OutFile $asc -UseBasicParsing ; \
+    Invoke-WebRequest $('https://yarnpkg.com/downloads/{0}/yarn-v{0}.tar.gz' -f $env:YARN_VERSION) -OutFile $tarGz -UseBasicParsing ; \
+    gpg --batch --verify $asc $tarGz
+
+ENV MHOLT_ARCHIVER_VERSION 2.0
+ENV MHOLT_ARCHIVER_SHA256 c23eab86a8f77e446f2b5dbb03305ac9acfb5187108a0090a4bcb3d908fa9785
+
+RUN Invoke-WebRequest $('https://github.com/mholt/archiver/releases/download/v{0}/archiver_windows_amd64.exe' -f $env:MHOLT_ARCHIVER_VERSION) -OutFile archiver.exe -UseBasicParsing ; \
```

Comment:
To extract the .tar.gz I used https://github.com/mholt/archiver which also provides a Windows binary. Added a checksum for the download.
This tool is simpler than using 7-Zip. WDYT?

Would also be great if someone gives PowerShell some more power to extract tar.gz natively.